docs/tutorial/connecting.rst

```diff
@@ -28,6 +28,11 @@ Then construct an object by passing the VISA address. For this example we connec
         from pymeasure.instruments import list_resources
         list_resources()
 
+    If you know the USB properties (vendor id, product id, serial numer) of the serial device, you can query for the VISA resource string::
+
+        from pymeasure.instruments import find_serial_port
+        resource_name = find_serial_port(vendor_id=15, product_id=0x12e5, serial_number="sn56X")
+
```

Comment:
Regarding USB, I understand vid and pid, but I am not sure whether serial number is a generic concept. Do you know what it means in usb terms?

Reply:
Almost all USB devices have these three hardware properties: a vendor id, a product id and a serial number. For USB-serial port adapters (as you find them for example in an Arduino or many other devices connected via USB but talking a serial protocol), the vendor is the producer of that adapter and these chips have a serial number (different of the device you talk to, for example a laser).

I know only one company not providing serial numbers for their devices. all other devices in our lab have serial_number.

My procedure is: connect the device, recognize the com port, then I read the hardware info (the ids and the serial number). Afterwards I can recognize that device by the hardware info and not the comport anymore.